tests/test_client.py

```diff
@@ -507,3 +507,117 @@ def test_wsgi_app(httpserver, monkeypatch):
   response = client.get("/")
   assert response.status_code == 200
   assert response.json == expected_response
+
+def test_scan_file_private(httpserver):
+    """Test synchronous private file scanning."""
+    upload_url = f"http://{httpserver.host}:{httpserver.port}/upload"
+
+    # Mock upload URL request
+    httpserver.expect_request(
+        "/api/v3/files/upload_url",
+        method="GET"
+    ).respond_with_json({
+        "data": upload_url
+    })
+
+    # Mock file upload with proper data structure
+    httpserver.expect_request(
+        "/upload", 
+        method="POST"
+    ).respond_with_json({
+        "data": {
+            "id": "dummy_scan_id",
+            "type": "analysis",
+            "links": {
+                "self": "dummy_link"
+            },
+            "attributes": {
+                "status": "queued",
+                "private": True,
+                "code_insight": True
+            }
+        },
+        "meta": {
+            "test": True
+        }
+    })
+
+    with new_client(httpserver) as client:
+        with io.StringIO("test file content") as f:
+            analysis = client.scan_file_private(f, code_insight=True)
+
+        assert analysis.id == "dummy_scan_id"
+        assert analysis.type == "analysis"
+        assert getattr(analysis, "status") == "queued"
+        assert getattr(analysis, "private") is True
+        assert getattr(analysis, "code_insight") is True
+
+@pytest.mark.asyncio 
+async def test_scan_file_private_async(httpserver):
+    """Test asynchronous private file scanning."""
+    upload_url = f"http://{httpserver.host}:{httpserver.port}/upload"
+
+    # Mock upload URL request
+    httpserver.expect_request(
+        "/api/v3/files/upload_url",
+        method="GET"
+    ).respond_with_json({
+        "data": upload_url
+    })
+
+    # Mock file upload
+    httpserver.expect_request(
+        "/upload",
+        method="POST"
+    ).respond_with_json({
+        "data": {
+            "id": "dummy_scan_id",
+            "type": "analysis",
+            "links": {
+                "self": "dummy_link"
+            },
+            "attributes": {
+                "status": "queued",
+                "private": True,
+                "code_insight": True
+            }
+        }
+    })
+
+    # Mock analysis status check
+    httpserver.expect_request(
+        "/api/v3/analyses/dummy_scan_id",
+        method="GET"
+    ).respond_with_json({
+        "data": {
+            "id": "dummy_scan_id",
+            "type": "analysis",
+            "links": {
+                "self": "dummy_link"
+            },
+            "attributes": {
+                "status": "completed",
+                "private": True,
+                "code_insight": True,
+                "stats": {
+                    "malicious": 0,
+                    "suspicious": 0
+                }
+            }
+        }
+    })
```

Comment:
This part of the code is shared between the previous test and this one, can you refactor so the code is not repeated?